### ISSUE TITLE
💅 Svelte : a component used as component AND type triggers the lint "All these imports are only used as types.biomelint/style/useImportType"

### ISSUE THREAD

**Issue by MarArMar** [biomejs/biome#3477]
### Environment information

```bash
bun
"@biomejs/biome": "^1.8.3",
6.5.0-10043-tuxedo x86_64 x86_64 x86_64 GNU/Linux
```

### Reproduction

- Create a Svelte component 
- Import it and use it both as type and as component
- See lint

MyComponent.svelte
```svelte
<div>My component !</div>
```

Case 1 : does not trigger the lint : 

+page.svelte
```svelte
<script lang="ts">
import MyComponent from './MyComponent.svelte'
</script>

<MyComponent />
```

Case 2 : triggers the lint : 

+page.svelte
```svelte
<script lang="ts">
import MyComponent from './MyComponent.svelte'

let myComponent: MyComponent
</script>

<MyComponent bind:this="{myComponent}" />
```

This 2nd syntax can be used to access component props & functions.

### Rule name

types.biomelint/style/useImportType

### Playground link

not needed

### Expected result

No lint

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
This was also reported in a discussion: #3426. I will close the issue as a duplicate of #3426.
